### ISSUE TITLE
Action slots on non-collapsible blocks have mechanism to tie back to heading for unique identifier

### ISSUE THREAD

**Issue by FelFly** [Esri/calcite-design-system#8037]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Summary

Per the Link Purpose guideline 2.4.4., I'm trying to avoid having multiple calcite-action buttons labeled the same exact thing, but given the actions are the same in several consecutive blocks and only the heading changes, I'd like to tie the actions to the heading via aria-labelledby for context.

### Actual Behavior

When I try to use aria-labelledby on the action slots in a non-collapsible block with a reference ID set on the heading of the block, I don't hear any type of relationship in NVDA or VoiceOver (it does work fine in JAWS). https://codepen.io/FelFly/pen/poxVNBv

### Expected Behavior

What I'd hope to hear is the title associated with the action slots, so for the remove buttons, I'd hear "remove, button, raccoons eating twizzlers" and "remove, button, all the puppies," which would help differentiate the buttons and provide more context when accessed from form shortcut menus for screen readers.

Kitty says this was fixed for collapsible blocks: https://github.com/Esri/calcite-design-system/pull/7652, so this issue is simply to extend the fix to non-collapsible blocks.

### Reproduction Sample

https://codepen.io/FelFly/pen/poxVNBv

### Reproduction Steps

1. In NVDA on Firefox, navigate into block
2. Navigate to the action links (open in new tab, remove) via tab for each of the two blocks

1. In VoiceOver, navigate into block (Ctrl+Option+right-arrow)
2. Navigate to action links (open in new tab, remove) via Ctrl+Option+right-arrow for each of the two blocks

### Reproduction Version

1.9.2

### Relevant Info

NVDA tested on Firefox in Windows 10 in Parallels, VoiceOver tested with Chrome (something about my codepen kept crashing Safari) on Mac Ventura 13.6.

### Priority impact

p3 - want for upcoming milestone

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Hub

**Comment by geospatialem** [comment]
Verified in `2.10.0-next.1` with https://codepen.io/FelFly/pen/poxVNBv and JAWS. 

However, Calcite is unable to provide support to NVDA, where NVDA and the shadow DOM conflict with `aria`-named attributes similar to what is outline in [this post](https://nolanlawson.com/2022/11/28/shadow-dom-and-accessibility-the-trouble-with-aria/).
